speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backend/FileDataStorage.cpp

```diff
@@ -98,64 +121,307 @@ void FileDataStorage::Shutdown()
+    WRITE_VAR(m_outputStream, GetLeastCommon(m_gearValues));  // gear
+}
+
+/// @brief Writes the human data from the last m_compressionRate time steps to the buffer file
+void FileDataStorage::WriteHumanData()
+{
+    WRITE_VAR(m_outputStream, GetMedian(m_steerValues));   // steer
+    WRITE_VAR(m_outputStream, GetMedian(m_brakeValues));   // brake
+    WRITE_VAR(m_outputStream, GetMedian(m_accelValues));   // gas
+    WRITE_VAR(m_outputStream, GetMedian(m_clutchValues));  // clutch
+}
+
+/// @brief Writes the intervention data from the last m_compressionRate time steps to the buffer file
+void FileDataStorage::WriteInterventionData()
+{
+    bool decisionMade = false;
+    WriteDecision(GetMedian(m_steerDecision), "SteerDecision", decisionMade);
+    WriteDecision(GetMedian(m_brakeDecision), "BrakeDecision", decisionMade);
+    WriteDecision(GetMedian(m_accelDecision), "AccelDecision", decisionMade);
+    WriteDecision(GetLeastCommon(m_gearDecision), "GearDecision", decisionMade);
+    WriteDecision(GetLeastCommon(m_lightDecision), "LightsDecision", decisionMade);
+
+    if (!decisionMade)
     {
-        WRITE_STRING(m_outputStream, "AccelDecision");
-        WRITE_VAR(m_outputStream, p_decisions.GetAccel());
+        WRITE_STRING(m_outputStream, "Decisions");
     }
-    if (p_decisions.ContainsGear())
+    WRITE_STRING(m_outputStream, "NONE");
+}
+
+/// @brief Writes the p_decisionType data from the last m_compressionRate time steps to the buffer file
+/// @param p_decision the value of the p_decisionType of the last time m_comrpessionRate time steps
+/// @param p_decisionType the current decision type
+/// @param p_decisionMade if there has already been a decision made
+template <typename TNumber>
+void FileDataStorage::WriteDecision(TNumber p_decision, const std::string& p_decisionType, bool& p_decisionMade)
+{
+    if (p_decision != -1)
     {
-        WRITE_STRING(m_outputStream, "GearDecision");
-        WRITE_VAR(m_outputStream, p_decisions.GetGear());
+        if (!p_decisionMade)
+        {
+            p_decisionMade = true;
+            WRITE_STRING(m_outputStream, "Decisions");
+        }
+        WRITE_STRING(m_outputStream, p_decisionType);
+        WRITE_VAR(m_outputStream, p_decision);
     }
-    if (p_decisions.ContainsLights())
+}
+
+/// @brief Add the new value to the total of the current time steps
+/// @param p_total The current total of the variable
+/// @param p_value The new value of this timestep for the variable
+void FileDataStorage::AddForAveraging(float& p_total, float p_value)
+{
+    p_total += p_value;
+}
+
+/// @brief Calculates the average of the past time steps for a variable and resets it
+/// @param p_total The current total of the variable
+/// @return The average of the past time steps for a variable
+float FileDataStorage::GetAverage(float& p_total) const
+{
+    float p_average = p_total / static_cast<float>(m_compressionRate);
+    p_total = 0;
+    return p_average;
+}
```

Comment:
This is not a parameter, so it should not be called p_average
```suggestion
float FileDataStorage::GetAverage(float& p_total) const
{
    float average = p_total / static_cast<float>(m_compressionRate);
    p_total = 0;
    return average;
}
```